sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/Metrics/ExecutableLinesWalkerTest.cs

```diff
@@ -328,5 +328,24 @@ static void Main(string[] args)
                 }",
                 4);
         }
+
+        [TestMethod]
+        public void Test_18_ExcludeFromTestCoverage()
+        {
+            AssertLinesOfCode(
+              @"using System.Diagnostics.CodeAnalysis;
+                namespace project_1
+                {
+                    public class ComplicatedCode
+                    {
+                        [ExcludeFromCodeCoverage]
+                        public string ComplexFoo()
+                        {
+                            string text = null;
+                            return text.ToLower();
+                        }
+                    }
+                }");

```

Comment:
It would be worth adding a non-excluded method afterwards which should have its lines counted (covers the situation raised by @Evangelink's comment above).